### ISSUE TITLE
Large gaps between windows on Win10

### ISSUE THREAD

**Issue by mowies** [jgpaiva/GridMove#10]
On Win10, I have really big gaps when i put windows side by side using gridmove. I think this happens due to the larger window borders in earlier Windows versions compared to Windows 10.

**Comment by Robbson** [comment]
Yes, this is something you see in all older programs of this kind, e.g. WinSplit Revolution, Power Resizer etc.

**Comment by mowies** [comment]
Is there a way to change the calculation of window borders in the program? I couldn't find it.

**Comment by Huseriato** [comment]
Did you find any solution?

**Comment by mowies** [comment]
No, unfortunately not.

**Comment by Huseriato** [comment]
This is annoying me since windows 10 is released. I've never found a better window manager than GridMove. I'm using it since first release. How do you deal with the problem? Are you using another one?

**Comment by mowies** [comment]
Yeah it's annoying me too. 
No still use GridMove and just adjust the windows by hand so that the gaps are filled. 
I looked into the issue a bit more but I can't seem to find where the corner positions of the windows get read out from Windows. 
It could of course also be an issue of Windows itself that the window positions are just off because of the new window design of Win10.

**Comment by Huseriato** [comment]
I've found a fixed offset in code:

```
        If not canceled
        {
          if(!AeroEnabled)
            WinMove,ahk_id %gui2hwnd%, ,%GridLeft%,%GridTop%,%GridWidth%,%GridHeight%
          else
          {
            left:=GridLeft + 3
            top:=GridTop + 3
            width:=GridWidth - 6
            height:=GridHeight - 6
            WinMove,ahk_id %gui2hwnd%, ,%Left%,%Top%,%Width%,%Height%
          }
        }
```

Some Windows (for example the window of the application "Telegram") are resized properly, so I think the !AeroEnabled flag is used here. May be it's window dependend.

**Comment by Huseriato** [comment]
I've tried to get it working. The GitHub version seems to be outdated. The helper.ahk is completely missing. The Grid-Loader has fixed templates inside (no file selection). The whole application is not running, when removing the helper/first run code. May be the auther has a new version or has lost its code and uploaded an old backup 4 years ago.

**Comment by Huseriato** [comment]
I've fixed it using the Aero Lite Theme in Windows 10. But I think it's no solution for everyone. http://winaero.com/blog/enable-the-hidden-aero-lite-theme-in-windows-10/

**Comment by mr-rak** [comment]
Fix WinMove function for Windows 10 https://github.com/mr-rak/GridMove

**Comment by Huseriato** [comment]
Wow, really cool. Thank you!

**Comment by mowies** [comment]
Thank you very much for the fix @mr-rak !

**Comment by tmuka** [comment]
I was annoyed by this issue too.  I've found making my grid areas 14 pixels wider and taller to be an acceptable workaround for me.  I subtract 7 pixels from the left, top, and add 7 to the right,bottom.

eg. this is the first part of my 4k_1366-160-874.grid
```
[1]                                                    
  TriggerTop    = [Monitor1Top] - 7                    
  TriggerRight   = [Monitor1Left] + 1366 + 7           
  TriggerBottom = [Monitor1Height] + 7                 
  TriggerLeft   = [Monitor1Left] -7                    
                                                       
[2]                                                    
  TriggerTop    = [Monitor1Top] - 7                    
  TriggerRight   = [Monitor1Left] + 1366 + 1600 +7     
  TriggerBottom = [Monitor1Height] + 7                 
  TriggerLeft   = [Monitor1Left] + 1366 -7             
                                                       
[3]                                                    
  TriggerTop    = [Monitor1Top] - 7                    
  TriggerRight   = [Monitor1Right] + 7                 
  TriggerBottom = [Monitor1Height] / 4 * 3 - 50 +7     
  TriggerLeft   = [Monitor1Left] + 1366 + 1600 -7      
```

**Comment by idvorkin** [comment]
@mr-rak Did you send a PR? It'd be great to have this merged back into the parent repo.

**Comment by VulcanTourist** [comment]
Apparently someone else will have to do that for @mr-rak, as he seems to have taken a sabbatical like the original GridMove author?

**Comment by mowies** [comment]
Yup. But his solution works for me at least

**Comment by VulcanTourist** [comment]
Yes, thankfully it does for me as well.  I am not fond of the alternatives to GridMove.

**Comment by jechin1** [comment]
How did you guys get mr-rak's fix to apply, it doesn't seem to change anything for me. I'm not very versed with AHK so if there is something special to do I may have missed it.

**Comment by mlgibbons** [comment]
Many thanks to Mr Rak for his Windows 10 fix.

**Comment by milynot** [comment]
> Fix WinMove function for Windows 10 https://github.com/mr-rak/GridMove

Thank you mr-rak.

**Comment by tmuka** [comment]
> Fix WinMove function for Windows 10 https://github.com/mr-rak/GridMove

Yes, thank you!  It's always fun when you setup a new computer and come across your own posts on the web about issues you encountered years ago!  I'm using the @mr-rak fix fork and it's working great. Thanks!

**Comment by idvorkin** [comment]
Check out fancy zones.  Similar idea and works great.  

https://github.com/microsoft/PowerToys/blob/master/src/modules/fancyzones

**Comment by tmuka** [comment]
> Check out fancy zones. Similar idea and works great.
> 
> https://github.com/microsoft/PowerToys/blob/master/src/modules/fancyzones

that seems like it has potential for drag and drop fans. I love gridmove for the grid position keyboard shortcuts!

**Comment by Huseriato** [comment]
> How did you guys get mr-rak's fix to apply, it doesn't seem to change anything for me. I'm not very versed with AHK so if there is something special to do I may have missed it.

Maybe a little late: You will need AutoHotKey installed (it's linked in his repo). Download mr-rak full repo and go into the `packaging`. There is a `compile.bat` - I don't know why, but the pathes are wrong. Replace with the following:

```
"C:\Program Files\AutoHotkey\Compiler\Ahk2Exe.exe" /in ../GridMove.ahk /out ../GridMove.exe /icon ../Images\GridMove.ico
```

And run the file. After that you'll have a GridMove.exe in the root-Directory of your downloaded GridMove copy of the mr-rak Repository. You can than use your grids as normal and run GridMove.exe.

Note: If you had overwritten your current installation of GridMove with the copy of mr-rak, than make sure, that you'd stopped GridMove before running compile.bat.

**Comment by GitMensch** [comment]
> Check out fancy zones. Similar idea and works great.
> 
> https://github.com/microsoft/PowerToys/blob/master/src/modules/fancyzones

That has a mandatory installation (no portable mode planned), see https://github.com/microsoft/PowerToys/issues/598#issuecomment-692483966

**Comment by Huseriato** [comment]
Why are you closing this? The problem is still there and not fixed at all.

**Comment by mowies** [comment]
The issue is so old already and the project doesn't seem to be maintained anymore. But I can also leave it open if that helps.

**Comment by endolith** [comment]
Is this why there is a gap between windows and the edge of the screen?
